Add support for Hash(ed) Time Lock(ed) Contracts (HTLC)

georgezgeorgez opened the pull request.

georgezgeorgez commented:
https://github.com/Big-Inches-Club-House/bich/discussions/1#discussioncomment-3341896

georgezgeorgez commented on vm/constants/embedded.go:
```diff
@@ -80,6 +81,14 @@ var (
 	SporkNameMaxLength        = 40
 	SporkDescriptionMaxLength = 400
 
+	/// === Htlc constants ===
```
I don't believe I am using these right now

georgezgeorgez commented on vm/embedded/tests/htlc_test.go:
```diff
@@ -0,0 +1,694 @@
+
+const (
+
+	// why not just make this pubic in mock genesis?
```
lol typo

coinselor replied:
Contrary to popular alien culture, there is nothing pubic about a Pillar and you should not be mocking the genesis block.

georgezgeorgez commented:
Awaiting SporkId to update and then merge

georgezgeorgez commented:
Marked as WIP to remove rpc list methods and possibly to add allowing other users to unlock and you receive funds

georgezgeorgez commented:
ready
just writing a ZIP now

Echo7seven commented:
Hello,   My apologies for the lateresponse.  

What were you referring to?   What are youready for?  What do I need to do ?

Again I apologize, I am new to everything .

 

 

 Oh NO !!

Saturday night I transferred a lot of ZNN to my newly acquiredSYRIUS  ZNN wallet.  It showed up inmy wallet so I was done, satisfied, happy I had accomplished it.  

But now my wallet shows a zero balance.   Please don’t tell me I messed up, and lost it.

 

If you don’t mind, please, explain what you were referringto in your email.   And what it is that you do so that I can learn andunderstand the processes.  

 

And if you can help me, or advise me about my ZNN wallet,that would be GREATLY appreciated.  

(i understand it may not be your job whatsoever, sosincerely,  THANK YOU )

 

 
                                                            THANK YOU

Clint Walkup





 
   On Sunday, February 19, 2023 at 06:40:42 PM EST, georgezgeorgez ***@***.***> wrote:  
 
 


ready
just writing a ZIP now

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

georgezgeorgez commented:
"Ready" was referring to the Pull Request for the HTLC code. You received an email because you are subscribed to updates for it. If you are having trouble with Syrius, I suggest you reach out to the community in Telegram or Discord, as that is unrelated to this PR.

georgezgeorgez commented:
Per CryptoFish suggestion, changed the rpc names to be consistent/shorter

georgezgeorgez commented:
fixed unlocks at the exact expiration time and added some tests for that edge case

ZIP is posted:
https://github.com/deeZNNutz-com/ZIP.deeZNNutz/blob/main/0002.md

beam us up

georgezgeorgez commented on vm/embedded/definition/htlc.go:
```diff
@@ -0,0 +1,244 @@
+}
+func unmarshalHtlcProxyUnlockInfoKey(key []byte) (*types.Address, error) {
+	if !isHtlcProxyUnlockInfoKey(key) {
+		return nil, errors.Errorf("invalid key! Not htcl proxy-unlock info key")
```
typo

georgezgeorgez commented on vm/embedded/definition/htlc.go:
```diff
@@ -0,0 +1,244 @@
+
+func unmarshalHtlcInfoKey(key []byte) (*types.Hash, error) {
+	if !isHtlcInfoKey(key) {
+		return nil, errors.Errorf("invalid key! Not htcl info key")
```
typo

zenon-network commented:
@georgezgeorgez the test `vm/embedded/implementation/htlc_test.go` fails with error `imports github.com/zenon-network/go-zenon/chain/genesis/mock: import cycle not allowed in test`

georgezgeorgez commented:
strange
it ran fine on my own branch
i'll investigate

georgezgeorgez commented:
```george@nom:~/repos/zenon-network/go-zenon$ git checkout master
Switched to branch 'master'
Your branch is behind 'origin/master' by 46 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)

george@nom:~/repos/zenon-network/go-zenon$ git pull
Updating c0f931d..82fc845
Fast-forward
[...]

george@nom:~/repos/zenon-network/go-zenon$ make znnd
go build -o /home/george/repos/zenon-network/go-zenon/build/znnd /home/george/repos/zenon-network/go-zenon/cmd/znnd/main.go
go: downloading github.com/ethereum/go-ethereum v1.10.22
go: downloading golang.org/x/sys v0.1.0
Build znnd done.
Run "/home/george/repos/zenon-network/go-zenon/build/znnd" to start znnd.

george@nom:~/repos/zenon-network/go-zenon$ go test vm/embedded/tests/htlc_test.go vm/embedded/tests/utils_test.go
ok  	command-line-arguments	1.907s

george@nom:~/repos/zenon-network/go-zenon$ go version
go version go1.18.1 linux/amd64

```

so during development i used `go test vm/embedded/tests/htlc_test.go vm/embedded/tests/utils_test.go` to do isolated testing

the github workflow file is using
`for s in $(go list ./...); do if ! go test -failfast -v -p 1 $s; then exit 1; fi; done`

i see the error in the pipeline
but what's strange is that when i try to run the same locally
my terminal just crashes out
looking at system resources, (i have 16 core, 32 gb ram on a very vanilla ubuntu)
it doesn't make a dent before it crashes

hmmm...

georgezgeorgez commented:
oh i guess it is the exit 1 and i am running it directly from my shell

georgezgeorgez commented:
oh yeah i just realized
the issue is with my htlc_test in the implementation package
which i put there along side token_test
it's using the mock
i'm just going to get rid of the dependency

georgezgeorgez commented:
@zenon-network 
okay sorry about that
oversight on my part
fixed
https://github.com/zenon-network/go-zenon/pull/18
